### PR TITLE
37346

### PR DESCRIPTION
**Pull request by sarathpyerra**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [37346](https://devops.dchbx.org/redmine/issues/37346) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | 2 |
| Peer Review Story Points | n/a |
| Ticket Author | Sarah Bagge |

### CODE REVIEW

File: app/models/services/checkbook_services.rb [hunk 1]
```diff
@@ -56,7 +56,9 @@ def csr_value
         active_house_hold = @hbx_enrollment.household.latest_active_tax_household_with_year(enrollment_year)
         if active_house_hold.nil?
           return "-01"
-        else 
+        elsif @hbx_enrollment.household.tax_households.tax_household_with_year(enrollment_year).active_tax_household.first.tax_household_members.flatten.select{|mem| mem.is_ia_eligible=="false"}.present?
```
Comment (Battula) [inline app/models/services/checkbook_services.rb]: Instead use `valid_csr_kind` on tax household.

File: app/models/services/checkbook_services.rb [hunk 2]
```diff
@@ -56,24 +56,27 @@ def csr_value
         active_house_hold = @hbx_enrollment.household.latest_active_tax_household_with_year(enrollment_year)
         if active_house_hold.nil?
           return "-01"
-        else 
-          case active_house_hold.latest_eligibility_determination.csr_percent_as_integer.to_s
-          when "100"
+        elsif @hbx_enrollment.household.tax_households.tax_household_with_year(enrollment_year).active_tax_household.first.tax_household_members.select{|mem| !mem.is_ia_eligible}.present? 
+          return "-01"
+        else
+          case active_house_hold.valid_csr_kind(hbx_enrollment)
```
Comment (saikumar9) [inline app/models/services/checkbook_services.rb]: Good usage of the **valid_csr_kind** method.

File: app/models/services/checkbook_services.rb [hunk 3]
```diff
@@ -56,24 +56,27 @@ def csr_value
         active_house_hold = @hbx_enrollment.household.latest_active_tax_household_with_year(enrollment_year)
         if active_house_hold.nil?
           return "-01"
-        else 
-          case active_house_hold.latest_eligibility_determination.csr_percent_as_integer.to_s
-          when "100"
+        elsif @hbx_enrollment.household.tax_households.tax_household_with_year(enrollment_year).active_tax_household.first.tax_household_members.select{|mem| !mem.is_ia_eligible}.present?
```
Comment (saikumar9) [inline app/models/services/checkbook_services.rb]: We might not need this **elseif** case at all, please remove this by verifying.

File: app/models/services/checkbook_services.rb [hunk 4]
```diff
@@ -56,24 +56,27 @@ def csr_value
         active_house_hold = @hbx_enrollment.household.latest_active_tax_household_with_year(enrollment_year)
```
Comment (saikumar9) [inline app/models/services/checkbook_services.rb]: Rename variables properly for better understanding.

File: spec/models/services/checkbook_services_spec.rb [hunk 5]
```diff
@@ -65,7 +65,9 @@ class ApplicationHelperModStubber
       it "should return correct csr percentage" do
         allow(tax_household).to receive(:eligibility_determinations).and_return [eligibility_determination_1]
         allow(hbx_enrollment).to receive_message_chain(:household,:latest_active_tax_household_with_year).and_return(tax_household)
-        expect(subject.csr_value).to eq "-05"
+        # allow(tax_household).to receive(:tax_households).and_return [eligibility_determination_1]
```
Comment (saikumar9) [inline spec/models/services/checkbook_services_spec.rb]: Please cover all the test cases.

File: app/models/services/checkbook_services.rb (hunk 3 above)
Comment (sarathpyerra) [reply app/models/services/checkbook_services.rb]: When Primary Family is Assisted and Child as Medicaid for this case we have to send default CSR as "-01” to Checkbook

Comment (saikumar9) [reply app/models/services/checkbook_services.rb]: Firstly, the above code snippet is not working as how you explained. Secondly, the method **valid_csr_kind** handles all the required cases including mixed household scenarios.

**Pull request merged**